taichi/codegen/codegen.h

```diff
@@ -22,6 +27,11 @@ class KernelCodeGen {
                                                Stmt *stmt = nullptr);
 
   virtual FunctionType codegen() = 0;
+  virtual LLVMCompiledData modulegen(
```

Comment:
Will this work properly when `TI_WITH_LLVM` is off?

Reply:
No, I forgot to add the #ifdef. However, IIUC all derived classes of KernelCodeGen use LLVM, then why do we only place this function inside #ifdef instead of place the whole KernelCodeGen inside it?

Reply:
lol if possible we can place the whole KernelCodeGen inside.

Reply:
Maybe we can move codegen.h/cpp into the `llvm/` directory after implementing the parallel compilation function. Let's just leave it here for now.